core/state_transition_ovm.go

```diff
@@ -0,0 +1,196 @@
+package core
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
+)
+
+var GodAddress = common.HexToAddress("0x444400000000000000000000000000000000000")
+var ZeroAddress = common.HexToAddress("0x0000000000000000000000000000000000000000")
+
+type ovmTransaction struct {
+	Timestamp     *big.Int       "json:\"timestamp\""
+	BlockNumber   *big.Int       "json:\"blockNumber\""
+	L1QueueOrigin uint8          "json:\"l1QueueOrigin\""
+	L1TxOrigin    common.Address "json:\"l1TxOrigin\""
+	Entrypoint    common.Address "json:\"entrypoint\""
+	GasLimit      *big.Int       "json:\"gasLimit\""
+	Data          []uint8        "json:\"data\""
+}
+
+func toExecutionManagerRun(evm *vm.EVM, msg Message) (Message, error) {
+	tx := ovmTransaction{
+		evm.Context.Time,
+		evm.Context.BlockNumber, // TODO (what's the correct block number?)
```

Comment:
@karlfloersch @tynes what's the correct value for this?

Reply:
For an L1 to L2 transaction, is `block.number` supposed to be the L1 block number?